File: fortran/witness_seed_3.0/tests/test_kairos.f90
program test_kairos
  use kairos, only: update_coherence
  use iso_fortran_env, only: real64
  implicit none
  real(real64) :: I(10), phase
  call random_number(I)
  phase = 0.1
  call update_coherence(I, phase)
  if (sum(abs(I)) > 0.0) then
    print *, "Kairos test passed: Coherence updated"
  else
    print *, "Kairos test failed"
    stop 1
  end if
end program test_kairos

File: haskell/witness_seed_3.0/tests/TestKairos.hs
module Main where

import System.Random
import Kairos

main :: IO ()
main = do
  g <- newStdGen
  let i = take 10 $ randoms g
      i' = coherence i 0.1
      sumAbs = sum $ map abs i'
  if sumAbs > 0
    then putStrLn "Kairos test passed: Coherence updated"
    else putStrLn "Kairos test failed" >> error "Test failed"